sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -584,14 +569,15 @@ public void uploadFromFileLease() throws IOException {
 
     @Test
     public void uploadFromFileLeaseFail() throws IOException {
-        primaryFileAsyncClient.create(DATA.getDefaultDataSizeLong()).block();
-        createLeaseClient(primaryFileAsyncClient).acquireLease().block();
         String fileName = generatePathName();
         String uploadFile = FileShareTestHelper.createRandomFileWithLength(1024, testFolder, fileName);
 
         StepVerifier
-            .create(primaryFileAsyncClient.uploadFromFile(uploadFile,
-                new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid())))
+            .create(primaryFileAsyncClient.create(DATA.getDefaultDataSizeLong())
+                .then(createLeaseClient(primaryFileAsyncClient).acquireLease())
+                .flatMap(leaseId -> primaryFileAsyncClient.uploadFromFile(uploadFile,
+                    new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid())))
+                .then(Mono.error(new ShareStorageException("Lease ID mismatch", null, null))))
```

Comment:
We don't want to manually throw an error, as that doesn't check if the error actually occurs when making the call. Check my previous comment to see how to change this.